diff --git a/asmcomp/arm/emit.mlp b/asmcomp/arm/emit.mlp
index cfcb0c9..3fd697e 100644
--- a/asmcomp/arm/emit.mlp
+++ b/asmcomp/arm/emit.mlp
@@ -579,9 +579,9 @@ let emit_item = function
   | Cint16 n ->
       `	.short	{emit_int n}\n`
   | Cint32 n ->
-      `	.word	{emit_nativeint	n}\n`
+      `	.word	{emit_nativeint	(Nativeint.logand n 0xffffffffn)}\n`
   | Cint n ->
-      `	.word	{emit_nativeint n}\n`
+      `	.word	{emit_nativeint (Nativeint.logand n 0xffffffffn)}\n`
   | Csingle f ->
       emit_float32_directive ".long" f
   | Cdouble f ->
