--- configure.ORIG	2011-10-19 18:23:18.000000000 +0200
+++ configure	2011-10-19 18:23:33.000000000 +0200
@@ -620,7 +620,7 @@
     *-*-cygwin*)                  natdynlink=true;;
     i[3456]86-*-linux*)           natdynlink=true;;
     x86_64-*-linux*)              natdynlink=true;;
-    i[3456]86-*-darwin10.*)
+    i[3456]86-*-darwin1[01].*)
       if test $arch64 == true; then
         natdynlink=true
       fi;;
